pkg/generate/generate.go

```diff
@@ -102,24 +108,26 @@ func Create(filepath string, name string) error {
 	var serviceText bytes.Buffer
 	serviceTemplate.Execute(&serviceText, replacement)
 
-	if err := ioutil.WriteFile(path.Join(filepath, "sample-component-builder.yaml"), []byte(componentBuilder), 0666); err != nil {
-		writeErr = err
+	compTmpl, _ := template.New("componentbuilder").Parse(componentBuilder)
+	var compText bytes.Buffer
+	compTmpl.Execute(&compText, replacement)
+
+	if err := ioutil.WriteFile(path.Join(dir, replacement.Name+"-builder.yaml"), compText.Bytes(), 0666); err != nil {
+		log.Exit(err)
 	}
-	if err := ioutil.WriteFile(path.Join(filepath, "sample-deployment.yaml"), deploymentText.Bytes(), 0666); err != nil {
-		writeErr = err
+	if err := ioutil.WriteFile(path.Join(dir, "sample-deployment.yaml"), deploymentText.Bytes(), 0666); err != nil {
+		log.Exit(err)
```

Comment:
with log.exit you're making assumptions about how this is used.  Ideally you should allow the calling context retain the ability to decide how to handle the error.

Reply:
Good point. I'll just return the error.

Reply:
I'll do that in a follow-up. I've got some more cleanups I was thinking about.